pyccel/ast/operators.py

```diff
@@ -1070,7 +1070,94 @@ class PyccelIsNot(PyccelIs):
     def __repr__(self):
         return '{} is not {}'.format(self.args[0], self.args[1])
 
+
 #==============================================================================
 
+class IfTernaryOperator(PyccelOperator):
+    """Represent a ternary conditional operator in the code, of the form (a if cond else b)
+
+    Parameters
+    ----------
+    args :
+        args : type list
+        format : condition , value_if_true, value_if_false
+
+    Examples
+    --------
+    >>> from sympy import Symbol
+    >>> from pyccel.ast.core import Assign
+	>>>	from pyccel.ast.operators import IfTernaryOperator
+    >>> n = Symbol('n')
+    >>> x = 5 if n > 1 else 2
+    >>> IfTernaryOperator(PyccelGt(n > 1),  5,  2)
+    IfTernaryOperator(PyccelGt(n > 1),  5,  2)
+    """
+    _precedence = 3
+
+    def __init__(self, cond, value_true, value_false):
+        super().__init__(cond, value_true, value_false)
+
+        if self.stage == 'syntactic':
+            return
+        if isinstance(value_true , Nil) or isinstance(value_false, Nil):
+            errors.report('None is not implemented for Ternary Operator', severity='fatal')
+        if isinstance(value_true , NativeString) or isinstance(value_false, NativeString):
+            errors.report('String is not implemented for Ternary Operator', severity='fatal')
+        if value_true.dtype not in NativeNumeric or value_false.dtype not in NativeNumeric:
+            errors.report('unkown data type in IfTernaryOperator', severity='fatal')
+        if value_true.dtype not in NativeNumeric:
+            assert(value_true.dtype == value_false.dtype)
```

Comment:
Please raise neat errors rather than raising assertions

Reply:
actually this should be removed, it doesn't make sense after the previous statement.

Reply:
That is because the previous statement is false

Reply:
You should have 3 cases:
1. `value_true.dtype in NativeNumeric  and value_false.dtype in NativeNumeric`
    **OK** : no error message (this may not need an if)
    Both values are numeric and are therefore compatible
    (e.g. `value_true = 4j`, `value_false = 4`)
2. `value_true.dtype == value_false.dtype`
    **OK** : no error message (this may not need an if)
    The values are non-numeric (as this is not case 1) but they are the same dtype so they are compatible
    (e.g. both values are instances of the same class)
3. Anything else
    **ERROR** : The types are incompatible, a clear error must be raised
    We don't know how to specify a dtype for different dtypes unless both dtypes are numeric so there is a problem

Reply:
great, i think this will fix #492 and #493, what do you think?

Reply:
No it won't. Those 2 issues are about fortran specific problems. The equivalent of the `IfTernaryOperator` in Fortran (`merge` ) cannot handle objects with different sizes. So in Fortran an `IfTernaryOperator` needs to be expanded into an `If` block in these 2 cases. This does nothing related to those issues